drivers/SmartThings/zigbee-window-treatment/src/mic/init.lua

```diff
@@ -0,0 +1,187 @@
+local window_shade_defaults = require "st.zigbee.defaults.windowShade_defaults"
+local WindowCovering = zcl_clusters.WindowCovering
+local cluster_base = require "st.zigbee.cluster_base"
+local utils = require "st.utils"
+local data_types = require "st.zigbee.data_types"
+local window_preset_defaults = require "st.zigbee.defaults.windowShadePreset_defaults"
+
+local micronic_window_shade_FINGERPRINTS = {{
+    mfr = "micronic-ko",
+    model = "acm301"
+}}
+
+local is_micronic_window_shade = function(opts, driver, device)
+    for _, fingerprint in ipairs(micronic_window_shade_FINGERPRINTS) do
+        if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+            return true
+        end
+    end
+    return false
+
+end
+
+local INVERT_CLUSTER = 0xFC00
+local INVERT_CLUSTER_ATTRIBUTE = 0x0000
+
+local function invert_preference_handler(device)
+
+    log.debug(string.format(" invert_preference_handler~~  (%s) invert(%s)", device.preferences, device.preferences.reverse))
+
+    local invert_value = device.preferences.reverse
+    local invert_cluster_cmd = cluster_base.write_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+        INVERT_CLUSTER_ATTRIBUTE, 0x0000, data_types.Boolean, invert_value)
+    device:send(invert_cluster_cmd)
+end
+
+local function info_changed(driver, device, event, args)
+    if device.preferences ~= nil and device.preferences.reverse ~= args.old_st_store.preferences.reverse then
+        invert_preference_handler(device)
+    end
+end
+
+local set_window_shade_level = function(level)
+
+    return function(driver, device, cmd)
+        device:send_to_component(cmd.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+    end
+end
+
+local WindowCovering = zcl_clusters.WindowCovering
+
+local SHADE_SET_STATUS = "shade_set_status"
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+
+    local level = value.value
+    local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+        capabilities.windowShadeLevel.shadeLevel.NAME)
+    local windowShade = capabilities.windowShade.windowShade
+
+    if level == 0 then
+        device:emit_event(windowShade.closed())
+        device:emit_event(capabilities.windowShadeLevel.shadeLevel(0))
+    elseif level == 100 then
+        device:emit_event(windowShade.open())
+        device:emit_event(capabilities.windowShadeLevel.shadeLevel(100))
+    else
+        if current_level ~= level or current_level == nil then
+            current_level = current_level or 0
+            device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+            local event = nil
+
+            if current_level ~= level then
+                --  event = current_level < level and windowShade.opening() or windowShade.closing()
+                event = current_level < level and windowShade.partially_open() or windowShade.partially_open()
```

Comment:
there's no reason to check the value here if you're going to emit the same event either way